src/validate.js

```diff
@@ -0,0 +1,183 @@
+let debug = require('debug')('taskcluster-lib-validate')
+let _ = require('lodash');
+let fs = require('fs');
+let path = require('path');
+let walk = require('walk')
+let yaml = require('js-yaml');
+let urljoin = require('url-join');
+let assert = require('assert');
+let Ajv = require('ajv');
+let AWS = require('aws-sdk');
+let Promise = require('promise');
+
+async function validator(options) {
+
+  let cfg = _.defaults(options, {
+    constants: './schemas/contants.yml',
+    folder: './schemas',
+    publish: process.env.NODE_ENV == 'production',
+    baseUrl: 'http://schema.taskcluster.net/',
+    bucket: 'schemas.taskcluster.net',
+  });
+
+  if (_.isString(cfg.constants)) {
+    let fullpath = path.resolve(cfg.constants);
+    debug('Attempting to set constants by file: %s', fullpath);
+    try {
+      cfg.constants = yaml.safeLoad(fs.readFileSync(fullpath, 'utf-8'));
+    }
+    catch (err) {
```

Comment:
nit, but I wouldn't make a newline... again I can't complain we have no linter :)